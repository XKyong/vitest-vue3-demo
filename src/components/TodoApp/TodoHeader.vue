<template>
  <header class="header">
    <h1>todos</h1>
    <input
      class="new-todo"
      placeholder="What needs to be done?"
      autofocus
      data-testid="new-todo"
      @keyup.enter="handleNewTodo"
    />
  </header>
</template>
<script setup lang="ts">

const emit = defineEmits(['new-todo'])

function handleNewTodo(e: KeyboardEvent) {
  const inputEl = e.target as HTMLInputElement
  const value = inputEl.value.trim()
  if (!value.length) return
  emit('new-todo', value)
  inputEl.value = ''
}
</script>
