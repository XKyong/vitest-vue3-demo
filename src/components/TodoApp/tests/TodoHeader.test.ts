import { shallowMount } from '@vue/test-utils'
import TodoHeader from '../TodoHeader.vue'

describe('TodoHeader.vue', () => {
  it('New todo', async () => {
    const wrapper = shallowMount(TodoHeader)
    const newTodoInput = wrapper.find('input[data-testid="new-todo"]')
    const text = 'play'
    await newTodoInput.setValue(text)
    await newTodoInput.trigger('keyup.enter')
    expect(wrapper.emitted()['new-todo']).toBeTruthy()
    // wrapper.emitted()['new-todo'] => [ [ 'play' ] ]
    expect((wrapper.emitted()['new-todo'][0] as string[])[0]).toBe(text)
    expect((newTodoInput.element as HTMLInputElement).value).toBe('')
  })
})
