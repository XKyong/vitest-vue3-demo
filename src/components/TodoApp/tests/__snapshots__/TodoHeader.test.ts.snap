// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoHeader.vue > snapshot 1`] = `
"<header class="header">
  <h1>todos</h1><input class="new-todo" placeholder="What needs to be done?" autofocus="" data-testid="new-todo">
</header>"
`;
